### PR TITLE
ci: stop inprogress jobs when pr is updated

### PR DESCRIPTION
**Pull request by Madhu-1**
**Description of your changes:**

stop the in-progress jobs when a PR is updated

Ref: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-concurrency-to-cancel-any-in-progress-job-or-run

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: .github/workflows/canary-integration-test.yml [hunk 1]
```diff
@@ -16,6 +16,11 @@ defaults:
     # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
     shell: bash --noprofile --norc -eo pipefail -x {0}
 
+# cancel the in-progress workflow when PR is refreshed.
```
Comment (travisn) [inline .github/workflows/canary-integration-test.yml:19]: What happens in the `push` case where multiple PRs are merged to the same branch (master or release branch)? Those tests should all run to completion. Hopefully only actions in PRs would be cancelled when the PR is updated, but multiple merges to a branch around the same time will not cancel.

**Review by travisn — CHANGES_REQUESTED** [review]

File: .github/workflows/integration-tests-on-release.yaml [hunk 2]
```diff
@@ -12,6 +12,11 @@ defaults:
     # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
     shell: bash --noprofile --norc -eo pipefail -x {0}
 
+# cancel the in-progress workflow when PR is refreshed.
```
[the remark this answers is not in the document]
Comment (Madhu-1) [reply .github/workflows/integration-tests-on-release.yaml:15]: Yes as this is for push event we don't need this one.

File: .github/workflows/canary-integration-test.yml (hunk 1 above)
Comment (Madhu-1) [reply .github/workflows/canary-integration-test.yml:19]: Yes AFAIU, this is applicable for the pull request update. i experimented on the pull requests.

Comment (subhamkrai) [reply .github/workflows/canary-integration-test.yml:19]: @Madhu-1 can we add condition something like this ```github.event_name != 'pull_request'```? or  this is not required

Comment (travisn) [reply .github/workflows/canary-integration-test.yml:19]: Looks like that's implemented now

**Review by travisn — APPROVED** [review]

**Pull request merged**